js/search.js

```diff
@@ -949,25 +949,38 @@
 			for (var i = 0; i < moves.length; i++) {
 				var id = moves[i];
 				var isViable = BattleMovedex[id] && BattleMovedex[id].isViable;
-				if (id === 'aerialace') isViable = (toId(set.ability) === 'technician');
+				if (id === 'aerialace') isViable = (toId(set.species) in {scyther:1, aerodactylmega:1, kricketune:1});
 				if (id === 'bellydrum') isViable = (toId(set.species) in {azumarill:1, linoone:1, slurpuff:1});
-				if (id === 'counter') isViable = (toId(set.species) in {chansey:1, skarmory:1, clefable:1, wobbuffet:1});
+				if (id === 'counter') isViable = (toId(set.species) in {chansey:1, skarmory:1, clefable:1, wobbuffet:1, alakazam:1});
+				if (id === 'curse') isViable = (toId(set.species) === 'snorlax');
+				if (id === 'drainingkiss') isViable = (toId(set.ability) === 'triage');
 				if (id === 'dynamicpunch') isViable = (toId(set.ability) === 'noguard');
+				if (id === 'electroball') isViable = (toId(set.species) === 'raichualola');
+				if (id === 'endeavor') isViable = (toId(set.species) in {mamoswine:1, scolipede:1});
 				if (id === 'focuspunch') isViable = (toId(set.species) === 'breloom');
 				if (id === 'gyroball') isViable = (template.baseStats.spe <= 60);
+				if (id === 'headbutt') isViable = (toId(set.species) === 'dunsparce');
 				if (id === 'hiddenpowerelectric') isViable = (moves.indexOf('thunderbolt') < 0);
 				if (id === 'hiddenpowerfighting') isViable = (moves.indexOf('aurasphere') < 0 && moves.indexOf('focusblast') < 0);
 				if (id === 'hiddenpowerfire') isViable = (moves.indexOf('flamethrower') < 0);
 				if (id === 'hiddenpowergrass') isViable = (moves.indexOf('energyball') < 0 && moves.indexOf('gigadrain') < 0);
 				if (id === 'hiddenpowerice') isViable = (moves.indexOf('icebeam') < 0 && template.id !== 'xerneas');
 				if (id === 'hypnosis') isViable = (this.gen < 4 && moves.indexOf('sleeppowder') < 0);
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'infestation') isViable = (toId(set.species) === 'shuckle');
+				if (id === 'irontail') isViable = (toId(set.species) in {absol:1, flygon:1, haxorus:1, lucario:1, mightyena:1, zygarde:1});
```

Comment:
This list looks long enough that you should be able to figure out a better rule for it, I think.

Reply:
(The problem with lists is that they usually don't count for LC and OMs that use middle evolutions, and you have to update them every new gen)

Reply:
is a two-part like steel type + !ironhead OR dark|dragon + !ironhead !gunkshot possible

the latter part would cover most everything on its own...except probably the most important user in luke

Reply:
Yeah, that works.

Reply:
Are you asking "possible" as in, you'd need my help to write the code? Or are you just asking if I'd support that, because I definitely consider that better than a list.

Reply:
i meant the latter but the former may well be true... how do you make "if" check the mon's type (a.k.a. what is the Pokemon type counterpart to toId(set.species) === 'magikarp')? 

After that, would it be this?
if (id === 'irontail') isViable = ((steeltypecheck && moves.indexOf('ironhead') < 0) || ((darktypecheck || dragontypecheck) && moves.indexOf('ironhead') < 0 && moves.indexOf('gunkshot') < 0));

Reply:
`template.types` contains types. Do an `indexOf` to check them.

And yes, that looks right to me.